WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomAmounts/AddCustomAmountPercentageViewModel.swift

```diff
@@ -16,13 +16,7 @@ final class AddCustomAmountPercentageViewModel: ObservableObject {
         return formattedAmount
     }
 
-    @Published var percentage = "" {
-        didSet {
-            guard oldValue != percentage else { return }
-
-            updateAmountBasedOnPercentage(percentage)
-        }
-    }
+    @Published var percentageTextFieldInput = ""
```

Comment:
Is it the input? I get that strictly it is, but it doesn't feel that way – the number comes from the text field.

I understand that the output comes via the `updatePercentageCalculatedAmount` now... but this name feels a little confusing, perhaps over-specific at some call sites.

Since it's intended for use in a binding, I'd expect the two way binding to work... but it doesn't unless you also use `updatePercentageCalculatedAmount`, which is unexpected. I can see that this only applies to `PercentageInputField`, which is a private implementation detail, so it's not a huge problem... but as a pattern I don't think a binding is correct if we can't rely on it to update the value itself.

```suggestion
    @Published private(set) var percentage = ""
```

This has other implications that would need to be worked through, and for private code it's a bit philosophical – let's merge as is to fix the bug. I wonder whether we should neaten it up this way as a follow up though?

Reply:
In this case, renaiming isn't needed, I could revert. I mostly did that here to make sure that I don't miss any usage of this variable and don't break any other behavior relying on this. 

I could've kept 'percentage'. It still serves as `input/output` for the text field, but no longer provides formatting of the value through `@Published var`.